src/args.rs

```diff
@@ -0,0 +1,47 @@
+#[derive(Debug, PartialEq, Eq)]
+pub struct Args {
+    pub command: String
+}
+
+impl Args {
+    pub fn parse(raw_args: Vec<String>) -> Result<Args, String> {
+        match raw_args.len() {
+            0 => Err(String::from("Please pass remote command.")), // TODO more user friendly message, for now it's consistent with Bash version.
+            _ => Ok(Args {
+                command: {
+                    let str: String = raw_args
+                        .iter()
+                        .cloned()
+                        // Add space between parameters.
+                        .map(|arg| format!("{} ", arg))
+                        .collect();
```

Comment:
... but better, I guess you are looking for something like Kotlin's `joinToString`. Rust also [have similar](https://doc.rust-lang.org/std/slice/trait.SliceConcatExt.html#tymethod.join).
It is unfortunate this function is not so discoverable.